lib/stages/BaseWriter.cpp

```diff
@@ -95,19 +96,24 @@ void BaseWriter::main_thread() {
 
     frameID frame_id(in_buf);
 
+    const int tsec = (int)acq_timeout;
+    const timespec timeout = {tsec, (long)((acq_timeout - tsec) * 1e9)};
+    int status;
```

Comment:
This should be defined within the loop to limit the scope, I would just do `auto status = wait_....`